tests/test_markers.py

```diff
@@ -394,3 +394,22 @@ def test_python_full_version_untagged_user_provided(self):
     def test_python_full_version_untagged(self):
         with mock.patch("platform.python_version", return_value="3.11.1+"):
             assert Marker("python_full_version < '3.12'").evaluate()
+
+    @pytest.mark.parametrize("variable", ["extras", "dependency_groups"])
+    @pytest.mark.parametrize(
+        "expression,result",
+        [
+            ('"foo" in {0}', True),
+            ('"bar" in {0}', True),
+            ('"baz" in {0}', False),
+            ('"baz" not in {0}', True),
+            ('"foo" in {0} and "bar" in {0}', True),
+            ('"foo" in {0} or "bar" in {0}', True),
+            ('"baz" in {0} and "foo" in {0}', False),
+            ('"foo" in {0} or "baz" in {0}', True),
+            ('"Foo" in {0}', True),
+        ],
+    )
+    def test_extras_and_dependency_groups(self, variable, expression, result):
+        environment = {variable: {"foo", "bar"}}
+        assert Marker(expression.format(variable)).evaluate(environment) == result
```

Comment:
nit: Let's use slightly more meaningful names in this test, like `value-one`/`value-two`/`value-not-in-set` or something along those lines.